src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -39,17 +43,27 @@ public AuthorizeFilter(AuthorizationPolicy policy)
         /// <param name="policyProvider">The <see cref="IAuthorizationPolicyProvider"/> to use to resolve policy names.</param>
         /// <param name="authorizeData">The <see cref="IAuthorizeData"/> to combine into an <see cref="IAuthorizeData"/>.</param>
         public AuthorizeFilter(IAuthorizationPolicyProvider policyProvider, IEnumerable<IAuthorizeData> authorizeData)
+            : this(authorizeData)
         {
             if (policyProvider == null)
             {
                 throw new ArgumentNullException(nameof(policyProvider));
             }
+
+            PolicyProvider = policyProvider;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="AuthorizeFilter"/>.
+        /// </summary>
+        /// <param name="authorizeData">The <see cref="IAuthorizeData"/> to combine into an <see cref="IAuthorizeData"/>.</param>
+        public AuthorizeFilter(IEnumerable<IAuthorizeData> authorizeData)
```

Comment:
Not super sure we need that. The motive for this bug wasn't to make it simplify the ctor but more about making it possible to add `AuthFilter` as a global filter.